### PR TITLE
ko: update repo location

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

ko has been moved out to its new repository named ko-build[^1], this PR aims to replace the ko's repository URL with the most recent one.

/cc @mattmoor @imjasonh 

[^1]: https://github.com/ko-build/ko/issues/791

### CODE REVIEW

**Comment by developer-guy** [comment]
AFAIK, no need to update the ldflag, PTAL @imjasonh

File: Formula/ko.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@ class Ko < Formula
   depends_on "go" => :build
 
   def install
-    system "go", "build", *std_go_args(ldflags: "-s -w -X github.com/google/ko/pkg/commands.Version=#{version}")
+    system "go", "build", *std_go_args(ldflags: "-s -w -X github.com/ko-build/ko/pkg/commands.Version=#{version}")
```
Comment (imjasonh) [inline Formula/ko.rb:22]:
```suggestion
    system "go", "build", *std_go_args(ldflags: "-s -w -X github.com/google/ko/pkg/commands.Version=#{version}")
```

For now, `ko`'s Go importpath is still `github.com/google/ko`, and we plan to move it to `ko.build` soonish. These paths are based on the importpath of the module, not the URL of the GitHub repo, so these need to be `google/ko` for now.

When we change the importpath and cut a release we'll want this to update to use the new importpath.

File: Formula/ko.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 class Ko < Formula
   desc "Build and deploy Go applications on Kubernetes"
-  homepage "https://github.com/google/ko"
-  url "https://github.com/google/ko/archive/v0.12.0.tar.gz"
+  homepage "https://github.com/ko-build/ko"
```
Comment (imjasonh) [inline Formula/ko.rb:3]:
```suggestion
  homepage "https://ko.build"
```

**Comment by chenrui333** [comment]
@developer-guy let's follow the commit style in here, https://docs.brew.sh/Formula-Cookbook#commit

**Comment by developer-guy** [comment]
it should be okay right now, thanks @chenrui333

**Review by chenrui333 — APPROVED** [review]

File: Formula/ko.rb (hunk 1 above)
Comment (imjasonh) [reply Formula/ko.rb]: This should still be Google for the time being

**Pull request closed without merging**